### DISCUSSION TITLE
Cannot figure out the correct `filter` query param syntax.

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#2858] [Q&A] [answered]
Hey there,  I have a collection called `users` which looks like the following:
```ts
{
  name: string,
  age: number
}
```
To get a list of `users` record, I tries to fetch the following url:
```
/api/collections/students/records?filter=(age<69 && name = 'whatever')
```
However, the response I got was
```json
{
  "code": 400,
  "message": "Invalid filter parameters.",
  "data": {}
}
```
Any idea on how to fix it?

**Comment by ganigeorgiev** [comment, marked as answer]
The above looks correct, but makes sure that the query parameter is properly URL encoded (the SDKs will do that automatically behind the scenes). For example, an equivalent version of the above url should be:

```
/api/collections/students/records?filter=%28age%3C69%20%26%26%20name%20%3D%20%27whatever%27%29
```

**Reply by Mr-M1M3** [reply]
How did you encode it? Any rules about which exact characters need to be encoded?

**Reply by Mr-M1M3** [reply]
Got it, you are using `encodeURIComponent`, I encoded it via `encodeURI` 😅😅😅
